varats-core/varats/provider/bug/bug.py

```diff
@@ -37,17 +38,31 @@ def introducing_commits(self) -> tp.List[pygit2.Commit]:
         return self.__introducing_commits
 
     @property
-    def issue_id(self) -> int:
-        """ID of the issue associated with the bug."""
+    def issue_id(self) -> tp.Optional[int]:
+        """ID of the issue associated with the bug, if there is one."""
         return self.__issue_id
 
+    def __eq__(self, other) -> bool:
+        if type(self) is type(other):
+            return (
+                self.fixing_commit == other.fixing_commit and
+                self.introducing_commits == other.introducing_commits and
```

Comment:
This only works for ordered lists.
Depending on the implementation of the introducing commit detection, we might want to also allow unordered lists to pass this check, e.g., something like this:
```suggestion
                set(self.introducing_commits) == set(other.introducing_commits) and
```